### ISSUE TITLE
Streams documentation for 1.0 doesn't link to 2.x equivalents

### ISSUE THREAD

**Issue by gmethvin** [akka/akka#20598]
For example, on http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/scala/stream-cookbook.html, the link at the bottom to the latest docs should link to http://doc.akka.io/docs/akka/2.4.6/scala/stream/stream-cookbook.html. Instead it just links back to the home page.

This is particularly annoying because only the older versions seem to appear on Google search for many search terms.

**Comment by ktoso** [comment]
Yeap I agree, would be good to fix... it involves some index perhaps... so a bit of pain to set it up.
Hopefully we'll be able to tell google somehow to stop directing to 1.0 hm

**Comment by richard-imaoka** [comment]
I found there are ways to [remove](https://support.google.com/webmasters/answer/6332384?hl=en) the old version from Google, particularly [block search indexing with meta tags](https://support.google.com/webmasters/answer/93710). 

> You can prevent a page from appearing in Google Search by including a noindex meta tag in the page's HTML code. When Googlebot next crawls that page, Googlebot will see the noindex meta tag and will drop that page entirely from Google Search results, regardless of whether other sites link to it.

However, if the newest version of the doc is not indexed by Google, the doc is completely gone from Google. When a new version of doc is deployed, does Akka notify Google to [add new URLs](https://www.google.com/webmasters/tools/submit-url?continue=/addurl), maybe via [PubSubHubbub](https://pubsubhubbub.appspot.com/)?

**Comment by ktoso** [comment]
Technically we always want people to look at "current" http://doc.akka.io/docs/akka/current/scala.html

And you'll notice we do this trick too:

```
<!-- Hint to search engines that the "canonical" page is under "current", which will boost it appearing in search results -->
<link rel="canonical" href="http://akka.io/docs/akka/current/scala.html" />		
```

Both good ideas Richard, let's see if we can use them :)

-- 
Konrad `ktoso` Malawski
Akka @ Lightbend

On 23 May 2016 at 21:22:06, Greg Methvin (notifications@github.com) wrote:

For example, on http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/scala/stream-cookbook.html, the link at the bottom to the latest docs should link to http://doc.akka.io/docs/akka/2.4.6/scala/stream/stream-cookbook.html. Instead it just links back to the home page.

This is particularly annoying because only the older versions seem to appear on Google search for many search terms.

**Comment by richard-imaoka** [comment]
It's best if Google can move the newest doc highest in its rank, but I guess that is probably difficult :|

I realized
* [http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/scala/stream-cookbook.html](http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/scala/stream-cookbook.html)

doesn't link to the same page on the newest version, but

* [http://doc.akka.io/docs/akka-stream-and-http-experimental/2.0.2/scala/http/common/xml-support.html](http://doc.akka.io/docs/akka-stream-and-http-experimental/2.0.2/scala/http/common/xml-support.html)

links to the same page on the newest. I'll see where the difference is from

**Comment by ktoso** [comment]
The difference is that we make a HEAD request to "the same url, but changed version to latest" and if the page exists, we show an additional link to direct people there. If a page has moved, or has been removed – which happened a few times between 1.0 and 2.0 and 2.4 then those links are not trivial to find magically.

-- 
Konrad `ktoso` Malawski
Akka @ Lightbend

On 25 May 2016 at 01:10:22, Richard Imaoka (notifications@github.com) wrote:

It's best if Google can move the newest doc highest in its rank, but I guess that is probably difficult :|

I realized

http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/scala/stream-cookbook.html
doesn't link to the same page on the newest version, but

http://doc.akka.io/docs/akka-stream-and-http-experimental/2.0.2/scala/http/common/xml-support.html
links to the same page on the newest. I'll see where the difference is from
